test/NuGet.Core.FuncTests/Dotnet.Integration.Test/DotnetSignTests.cs

```diff
@@ -0,0 +1,497 @@
+                result.Success.Should().BeFalse(because: result.AllOutput);
+                //result.AllOutput.Should().Contain(string.Format(_invalidPasswordErrorCode, pfxPath));
+                result.AllOutput.Should().Contain(_invalidPasswordErrorCode);
+            }
+        }
+
+        [Fact]
+        public async Task DotnetSign_SignPackageWithUntrustedSelfIssuedCertificateInCertificateStore_SuccessAsync()
+        {
+            // Arrange
+            using (var pathContext = _msbuildFixture.CreateSimpleTestPathContext())
+            {
+                await SimpleTestPackageUtility.CreatePackagesAsync(
+                    pathContext.PackageSource,
+                    new SimpleTestPackageContext("PackageA", "1.0.0"));
+
+                var packageFilePath = Path.Combine(pathContext.PackageSource, "PackageA.1.0.0.nupkg");
+
+                var untrustedSelfIssuedCert = _signFixture.UntrustedSelfIssuedCertificateInCertificateStore;
+
+                //Act
+                var result = _msbuildFixture.RunDotnet(
+                    pathContext.PackageSource,
+                    $"nuget sign {packageFilePath} --certificate-fingerprint {untrustedSelfIssuedCert.Thumbprint}",
+                    ignoreExitCode: true);
+
+                // Assert
+                result.Success.Should().BeTrue(because: result.AllOutput);
+                result.AllOutput.Should().Contain(_noTimestamperWarningCode);
+            }
+        }
+
+        [Fact]
+        public async Task DotnetSign_SignPackageWithUnsuportedTimestampHashAlgorithm_FailsAsync()
+        {
+            // Arrange
+            using (var pathContext = _msbuildFixture.CreateSimpleTestPathContext())
+            {
+                await SimpleTestPackageUtility.CreatePackagesAsync(
+                    pathContext.PackageSource,
+                    new SimpleTestPackageContext("PackageA", "1.0.0"));
+
+                var packageFilePath = Path.Combine(pathContext.PackageSource, "PackageA.1.0.0.nupkg");
+                var originalFile = File.ReadAllBytes(packageFilePath);
+
+                var testServer = await _signFixture.GetSigningTestServerAsync();
+                var certificateAuthority = await _signFixture.GetDefaultTrustedCertificateAuthorityAsync();
+                var options = new TimestampServiceOptions() { SignatureHashAlgorithm = new Oid(Oids.Sha1) };
+                var timestampService = TimestampService.Create(certificateAuthority, options);
+
+                using (testServer.RegisterResponder(timestampService))
+                using (var UntrustedSelfIssuedCert = _signFixture.UntrustedSelfIssuedCertificateInCertificateStore)
+                {
+                    //Act
+                    var result = _msbuildFixture.RunDotnet(
+                        pathContext.PackageSource,
+                        $"nuget sign {packageFilePath} --certificate-fingerprint {UntrustedSelfIssuedCert.Thumbprint} --timestamper {timestampService.Url}",
```

Comment:
If CertificatePath is specified, the cert will be from a certificate file.
If CertificateFingerprint/CertificateSubjectName/CertificateStoreLocation/CertificateStoreName is specified, it will search for a local certificate store.
If both of the two categories are specified, we will throw an exception.

If it's from local certificate store,
the default StoreName is StoreName.My.
the default StoreLocation is StoreLocation.CurrentUser.
If StoreName/StoreLocation is not specified, they will use the default value to search.
If the cert is not in My\CurrentUser, you have to specify the StoreName and StoreLocation.